pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -302,29 +314,35 @@ func (facts *PipelineRunFacts) IsGracefullyStopped() bool {
 
 // DAGExecutionQueue returns a list of DAG tasks which needs to be scheduled next
 func (facts *PipelineRunFacts) DAGExecutionQueue() (PipelineRunState, error) {
-	tasks := PipelineRunState{}
-	// when pipeline run is stopping, gracefully cancelled or stopped, do not schedule any new task and only
-	// wait for all running tasks to complete and report their status
-	if !facts.IsStopping() && !facts.IsGracefullyCancelled() && !facts.IsGracefullyStopped() {
-		// candidateTasks is initialized to DAG root nodes to start pipeline execution
-		// candidateTasks is derived based on successfully finished tasks and/or skipped tasks
-		candidateTasks, err := dag.GetSchedulable(facts.TasksGraph, facts.successfulOrSkippedDAGTasks()...)
-		if err != nil {
-			return tasks, err
-		}
+	var tasks PipelineRunState
+	// when pipelinerun is gracefully cancelled or gracefully stopped, do not schedule any new tasks,
+	// and only wait for all running tasks to complete (without exhausting retries).
+	if facts.IsGracefullyCancelled() || facts.IsGracefullyStopped() {
+		return tasks, nil
+	}
```

Comment:
we've documented that retries are not executed in (non-graceful) cancellation as well, why is that not included in the conditions here? 🤔

Reply:
I think it was never included because it's handled elsewhere-- however it makes sense for this to not return any tasks if the pipeline is cancelled so I've added that in (plus a test case)

Reply:
sounds good, thanks :)